### PR TITLE
FEATURE: FET-1148 Conventional commits

### PR DESCRIPTION
**Pull request by kandl**
- from now on, commit messages should conform to
  [conventional commits](https://www.conventionalcommits.org/) specification
- update contribution guide
- add github action for pull request title validation
- add github action for commit messages validation
- 
JIRA: FET-1148

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: .github/actions/check-pull-request-commits/commitlint.config.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// (C) 2022 GoodData Corporation
+
+const LINE_WIDTH = 72;
```
Comment (kandl) [inline .github/actions/check-pull-request-commits/commitlint.config.js:3]: I'm not sure if 72 characters per line is enough, thoughts?

File: .github/actions/check-pull-request-commits/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": "check-pull-request-commits",
+    "version": "1.0.0",
+    "description": "Simple GitHub action that checks pull request commits",
+    "main": "src/action.js",
+    "author": "GoodData",
+    "license": "UNLICENSED",
```
Comment (kandl) [inline .github/actions/check-pull-request-commits/package.json:7]: I'm not sure if `UNLICENSED` is correct

File: .github/actions/check-pull-request-title/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "check-pull-request-title",
+    "version": "1.0.0",
+    "description": "Simple GitHub action that checks pull request title",
+    "main": "index.js",
+    "scripts": {
+        "build": "ncc build src/action.ts -o build"
+    },
+    "author": "GoodData",
+    "license": "UNLICENSED",
```
Comment (kandl) [inline .github/actions/check-pull-request-title/package.json:10]: I'm not sure if `UNLICENSED` is correct

File: .github/actions/check-pull-request-title/src/action.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    \`CHANGE_TYPE: JIRA_TICKET_ID description\`, or
+    \`CHANGE_TYPE: JIRA_TICKET_ID1, JIRA_TICKET_ID2 description\`, or
+    \`CHANGE_TYPE: description\` for trivial changes
+
+CHANGE_TYPE can be:
+    - FEATURE
+    - BUGFIX
+    - RELATED
+    - TRIVIAL
+    - CONFIG
```
Comment (kandl) [inline .github/actions/check-pull-request-title/src/action.ts:31]: Does this categorization make sense in SDK?
Build and tooling changes under `FEATURE` flag may be misleading?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: .github/actions/check-pull-request-title/src/action.ts (hunk 4 above)
Comment (no23reason) [reply .github/actions/check-pull-request-title/src/action.ts:31]: I personally avoid it, as some of them automatically close the related ticket which is almost never what I want. So I always use RELATED: it does not have any magic behind it.

File: .github/actions/check-pull-request-title/package.json (hunk 3 above)
Comment (no23reason) [reply .github/actions/check-pull-request-title/package.json:10]: That is a very good question :-/ would MIT be better? we do not use any non-MIT stuff here, right?

File: .github/actions/check-pull-request-commits/commitlint.config.js (hunk 1 above)
Comment (no23reason) [reply .github/actions/check-pull-request-commits/commitlint.config.js:3]: I think it is fine, personally I stick with what VSCode suggests:
50 for the title, 72 for body.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/10411/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12409/ : SUCCESS in 21m 29s

File: .github/actions/check-pull-request-title/package.json (hunk 3 above)
Comment (kandl) [reply .github/actions/check-pull-request-title/package.json:10]: We don't. Let's change it to MIT.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by hackerstanislav — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3168

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/45/ : FAILURE in 2m 01s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3311/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2159/ : SUCCESS in 21m 44s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12418/ : SUCCESS in 21m 01s

**Pull request merged**